src/parser/shaman/elemental/modules/talents/SurgeOfPower.js

```diff
@@ -0,0 +1,118 @@
+import React from 'react';
+
+import Analyzer from 'parser/core/Analyzer';
+import StatisticBox, { STATISTIC_ORDER } from 'interface/others/StatisticBox';
+import SpellIcon from 'common/SpellIcon';
+import SPELLS from 'common/SPELLS';
+import SpellLink from 'common/SpellLink';
+import { SELECTED_PLAYER } from 'parser/core/EventFilter';
+import Events from 'parser/core/Events';
+
+const SURGE_OF_POWER = {
+  DURATION: 15000,
+  WINDOW_DURATION: 500,
+  AFFECTED_CASTS: [
+    SPELLS.FLAME_SHOCK,
+    SPELLS.FROST_SHOCK,
+    SPELLS.LAVA_BURST,
+    SPELLS.LIGHTNING_BOLT,
+  ],
+};
+
+class SurgeOfPower extends Analyzer {
+  sopBuffedAbilities = {};
+  // % of SK lightning bolts empoered by SoP
+  sopSKCastsPercentage = 0.0;
```

Comment:
I would say either remove it and the commented console log or set it somewhere, introduce a debug variable and do (debug && console.log) in line 94.